### PR TITLE
Dynamic tax rules

### LINKED ISSUE #52: Add "dynamic" rules on the fly for taxes calculation
Today the problem is very static and sometimes people tend to implement it "from one shot". Furthermore, this approach and the /feedback resources don't push participants to test.

After many feedbacks and discussions with others running Extreme Carpaccio, we're willing to add new taxes rules on the fly to push the problem nature from static (at the beginning) to dynamic (during the game).

To do so, at some point we'd like to change the taxes strategy leaving the fixed numbers back and start using rules for each country, like : `if(total < 100) return sum * 1.2; else return sum * 1.2 + 10;`

### PR DESCRIPTION
**Pull request by Arnauld**
Resolve #52 - allow to dynamically change tax rule calculation based on configuration.
Configuration allow scale factor or function definition.

Configuration example:

```json
{
   ...
   
  "taxes": {
    "SK": "function(price) { if(price>2000) return price * 1.2; else return price * 1.1; }",
    "UK": 1.45
  }
}
```

### CODE REVIEW

File: server/specs/repositories_spec.js [hunk 1]
```diff
@@ -149,5 +154,45 @@ describe('Countries', function() {
...
+        spyOn(configuration, 'all').andReturn({taxes: {
+            'CY': "function(price) { return 1234; }"
+        }});
+
+        var newTax = countries.taxRule('CY');
+
+        expect(newTax.applyTax(421)).toBe(1234);
+    });
+
+    it('should keep tax unchanged when it fails to read it from configuration - invalid function case', function() {
```
Comment (dlresende) [inline server/specs/repositories_spec.js]: Corner case covered! Perfect :)

**Comment by dlresende** [comment]
@Arnauld: it seems the following test case is failing:
```
  1) Countries should keep tax unchanged when it fails to execute evaluation from configuration
   Message:
     Expected 7 to be 281.82.
```
What do you mean by _fails to execute evaluation from configuration_? Is that when the syntax is not correct?

**Comment by Arnauld** [comment]
Function should have thrown an error:

```javascript
it('should keep tax unchanged when it fails to execute evaluation from configuration', function() {
        spyOn(configuration, 'all').andReturn({taxes: {
            'EE': "function(price) {throw {error:'oops'};}"
        }});

        expect(countries.taxRule('EE').applyTax(231)).toBe(231 * 1.22);
    });
```

I had issue in the meanwhile and forgot to fix it 🍶
